src/sbvr-api/permissions.ts

```diff
@@ -1660,7 +1661,8 @@ export const addPermissions = async (
 	req: PermissionReq,
 	request: ODataRequest & { permissionType?: PermissionCheck },
 ): Promise<void> => {
-	const { vocabulary, resourceName, odataQuery, odataBinds } = request;
+	const { resourceName, odataQuery, odataBinds } = request;
+	const vocabulary = _.last(request.translateVersions)!;
```

Comment:
It is always initialized with at least one element as
```ts
const versions = [apiRoot];
```
(which is then assigned into the right place), which means there will always be at least 1 element so `_.last` will always have a string element to return. The `T | undefined` typing of `_.last` is because when there array has 0 elements it'll return `undefined`, if you can guarantee `length >= 1` then the typing should be just `T`

Reply:
I realized there's another part of this which is that we don't guarantee that the object being passed in has to be one that has undergone the addition of eg engines/translateVersions/etc and therefore `translateVersions` could be missing in that case, if that were to happen it would fail fast and presumably be fairly straightforward to fix but I've opened https://github.com/balena-io/pinejs/pull/565 as a way to differentiate between the "just parsed" vs the amended/"processed" request so that we can enforce that it is defined